packages/server/src/sdk/app/backups/imports.ts

```diff
@@ -39,44 +62,45 @@ async function updateAttachmentColumns(prodAppId: string, db: Database) {
     if (attachmentCols.length === 0) {
       continue
     }
-    // use the CouchDB Mango query API to lookup rows that have attachments
-    const params: CouchFindOptions = {
-      selector: {
-        _id: {
-          $regex: `^${TABLE_ROW_PREFIX}`,
-        },
-      },
-    }
-    attachmentCols.forEach(col => (params.selector[col] = { $exists: true }))
-    const { rows } = await dbCore.directCouchFind(db.name, params)
-    for (let row of rows) {
-      for (let column of attachmentCols) {
-        if (!Array.isArray(row[column])) {
-          continue
-        }
-        row[column] = row[column].map((attachment: RowAttachment) => {
-          // URL looks like: /prod-budi-app-assets/appId/attachments/file.csv
-          const urlParts = attachment.url.split("/")
-          // drop the first empty element
-          urlParts.shift()
-          // get the prefix
-          const prefix = urlParts.shift()
-          // remove the app ID
-          urlParts.shift()
-          // add new app ID
-          urlParts.unshift(prodAppId)
-          const key = urlParts.join("/")
-          return {
-            ...attachment,
-            key,
-            url: `/${prefix}/${key}`,
+    let bookmark: null | string = null,
+      rowsLength = 0
+    do {
+      const params = generateAttachmentFindParams(attachmentCols, bookmark)
+      // use the CouchDB Mango query API to lookup rows that have attachments
+      const resp = await dbCore.directCouchFind(db.name, params)
+      bookmark = resp.bookmark
+      rowsLength = resp.rows.length
+      const rows = resp.rows
+      for (let row of rows) {
+        for (let column of attachmentCols) {
+          if (!Array.isArray(row[column])) {
+            continue
           }
-        })
+          row[column] = row[column].map((attachment: RowAttachment) => {
+            // URL looks like: /prod-budi-app-assets/appId/attachments/file.csv
+            const urlParts = attachment.url.split("/")
+            // drop the first empty element
+            urlParts.shift()
+            // get the prefix
+            const prefix = urlParts.shift()
+            // remove the app ID
+            urlParts.shift()
+            // add new app ID
+            urlParts.unshift(prodAppId)
+            const key = urlParts.join("/")
+            return {
+              ...attachment,
+              key,
+              url: `/${prefix}/${key}`,
+            }
+          })
+        }
```

Comment:
Could we pull the mapping out into a a function with a dedicated name?
It'd be a bit easier to see the intended purpose then like:

```ts
reWriteAttachmentUrl(appId: string, attachment: RowAtachment): RowAttachment
```

Reply:
There's an alternative we can make to this sort of behaviour, where we use the app id in context to dynamically enrich the url, instead of embedding the app id into the row itself - something we might consider in future.

Reply:
Utilising the context in this case could be a little dangerous, as it is around the creation of the app - the ID may not yet be in context as the app doesn't really exist yet.

Reply:
Ah no for this I mean on read. So we never store the appId in the attachment url, we can enrich it on read, like we do for tenancy and presigned urls